### PR TITLE
Adds support for the Address fields on Organisation

### PR DESCRIPTION
**Pull request by nikz**

### CODE REVIEW

File: lib/xero_gateway/base_record.rb [hunk 1]
```diff
@@ -59,34 +62,50 @@ def from_xml_attributes(element, attribute = nil, attr_definition = self.class.a
...
+
     def to_xml_attributes(builder = Builder::XmlMarkup.new, path = nil, attr_definitions = self.class.attribute_definitions)
       attr_definitions.each do |attr, value|
         case value
         when Hash
           builder.__send__(attr) do
             to_xml_attributes(builder, "#{path}#{attr}", value)
           end
+        when Array
+          raise UnsupportedAttributeType.new("#{value} instances don't respond to #to_xml") unless value.first.method_defined?(:to_xml)
```
Comment (armstrjare) [inline lib/xero_gateway/base_record.rb]: what if empty?

File: test/unit/organisation_test.rb [hunk 2]
```diff
@@ -14,14 +14,47 @@ def test_build_and_parse_xml
...
+
+  test "should raise if you add an unsupported type as an address" do
+    org = create_test_organisation
+    org.addresses = [ { :a => 123 }]
+
+    assert_raises "UnsupportedAttributeType" do
+      org.to_xml
+    end
+  end
+
```
Comment (armstrjare) [inline test/unit/organisation_test.rb]: pretty sure current implementation will raise if `org.addresses = []` ?

File: lib/xero_gateway/base_record.rb (hunk 1 above)
Comment (nikz) [inline lib/xero_gateway/base_record.rb]: Are we going to support not specifying the content of the array?

File: test/unit/organisation_test.rb (hunk 2 above)
Comment (nikz) [inline test/unit/organisation_test.rb]: Have added a test, not sure it does but maybe I'm missing the scenario you're thinking of?

Comment (armstrjare) [inline test/unit/organisation_test.rb]: Maybe it works because `nil` resonds to `to_xml` ?

File: lib/xero_gateway/base_record.rb (hunk 1 above)
Comment (armstrjare) [inline lib/xero_gateway/base_record.rb]: I think it works because `nil.respond_to?(:to_xml) == true`

**Pull request merged**